packages/frontend/src/App.jsx

```diff
@@ -1,28 +1,39 @@
 import React, { useEffect } from 'react';
 import './styles/App.scss';
 import { Routes, Route, useNavigate } from 'react-router-dom';
-import { routes } from './utils';
+import { routes, GAME_STAGES } from './utils';
 import useGameContext from 'hooks/useGameContext';
+import useFormContext from 'hooks/useFormContext';
 
 const App = () => {
   const navigate = useNavigate();
   const {
     state: { stage },
   } = useGameContext();
 
+  const {
+    state: { currentStep, formValues },
+  } = useFormContext();
+
   useEffect(() => {
     const { path } = routes[stage];
-    navigate(path);
+    let pathEdit = path.slice(0, path.length - 2);
+
+    let pathStep = routes[stage].children
+      ? pathEdit.concat(routes[stage].children[currentStep - 1].path)
+      : '';
+
+    navigate(stage === GAME_STAGES.SETUP ? pathStep : path);
   }, [stage]);
```

Comment:
I feel like the path navigation should be based on the location that's entered in the browser. React Router has a `useLocation` hook which returns a `pathname` value which can then be split and analyzed. This will be useful if someone enters `/setup/foo`. There isn't a step that matches that path, so we could look at the `currentStep` value in the form state and determine which step should be rendered.

Reply:
So we would need to manually check if the incoming location matches any or our possible locations by splitting it into two separate strings and checking that the first half matches one of our parent routes and the second matches one of its child routes. If both of those match then we do nothing because the entered route is a good one, but if it doesn't match then we used the forced navigate to the path as constructed by the data in the context?